pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/text/NativeTextIndexReader.java

```
@@ -0,0 +1,141 @@
+
+  private FST _fst;
+  private BitmapInvertedIndexReader _invertedIndex;
+
+  public NativeTextIndexReader(String column, File indexDir, int numDocs) {
+    _column = column;
+    _numDocs = numDocs;
+    try {
+      _indexFile = getTextIndexFile(indexDir);
+      _buffer = PinotDataBuffer.loadBigEndianFile(_indexFile);
+
+      populateIndexes();
+    } catch (Exception e) {
+      LOGGER.error("Failed to instantiate Lucene text index reader for column {}, exception {}", column,
+          e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  private File getTextIndexFile(File segmentIndexDir) {
+    // will return null if file does not exist
+    File file = SegmentDirectoryPaths.findTextIndexIndexFileNative(segmentIndexDir, _column);
+    if (file == null) {
+      throw new IllegalStateException("Failed to find text index file for column: " + _column);
+    }
+    return file;
+  }
+
+  private void populateIndexes() {
+    long invertedIndexLength = _buffer.getLong(4);
+    long fstDataLength = _buffer.getLong(12);
+    int numBitMaps = _buffer.getInt(20);
+
+    long fstDataStartOffset = NativeTextIndexCreator.HEADER_LENGTH;
+    long fstDataEndOffset = fstDataStartOffset + fstDataLength;
+    ByteBuffer byteBuffer = _buffer.toDirectByteBuffer(fstDataStartOffset, (int) fstDataLength);
+    byte[] arr = new byte[byteBuffer.remaining()];
+    // byteBuffer.get(arr);
+    try {
+      _fst = FST.read(new ByteBufferInputStream(Collections.singletonList(byteBuffer)), ImmutableFST.class, true);
+    } catch (IOException e) {
+      throw new RuntimeException(e.getMessage());
+    }
+
+    long invertedIndexEndOffset = fstDataEndOffset + invertedIndexLength;
+    _invertedIndex =
+        new BitmapInvertedIndexReader(_buffer.view(fstDataEndOffset, invertedIndexEndOffset, ByteOrder.BIG_ENDIAN),
+            numBitMaps);
+  }
+
+  @Override
+  public ImmutableRoaringBitmap getDictIds(String searchQuery) {
+    try {
```

Should throw `UnsupportedException` here as we should never ask for dictionary ids from text index (the returned dictionary id is not the dictionary id for the column, but for the tokens)